docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst

```diff
@@ -0,0 +1,266 @@
+Assuming you have a pod with two containers, one is a producer and the other is a consumer.
+The producer will generate some data and write it to vineyard, and the consumer will read the data
+from vineyard and do some computation.
+
+Save the following yaml as `pod.yaml`.
+
+.. code:: yaml
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: vineyard-showcase
+      namespace: vineyard-test
+    spec:
+      containers:
+      - name: producer
+        image: python:3.10
+        command: ["/bin/sh", "-c", "pip install --user -q --disable-pip-version-check vineyard numpy pandas; python -c \"$put_dataframe\"; python -c \"$put_basic_data_unit\"; sleep infinity"]
+        env:
+          - name: put_dataframe
+            value: |
+              import vineyard;
+              import numpy as np;
+              import pandas as pd;
+              client = vineyard.connect();
+              client.put(pd.DataFrame(np.random.randn(100, 4), columns=list('ABCD')), persist=True, name="test_dataframe");
+              client.close()
+          - name: put_basic_data_unit
+            value: |
+              import vineyard;
+              import numpy as np;
+              client = vineyard.connect();
+              client.put((1, 1.2345, 'xxxxabcd'), persist=True, name="test_basic_data_unit");
+              client.close()
+      - name: consumer
+        image: python:3.10
+        command: ["/bin/sh", "-c", "pip install --user -q --disable-pip-version-check vineyard numpy pandas; python -c \"$get_dataframe\"; python -c \"$get_basic_data_unit\"; sleep infinity"]
+        env:
+          - name: get_dataframe
+            value: |
+              import vineyard;
+              client = vineyard.connect();
+              print(client.get(name="test_dataframe").sum())
+              client.close()
+          - name: get_basic_data_unit
+            value: |
+              import vineyard;
+              client = vineyard.connect();
+              print(client.get(name="test_basic_data_unit"))
+              client.close()
+
+Use the `vineyardctl` to inject vineyard into the pod and apply them to the kubernetes cluster
+as follows.
+
+.. code:: bash
+
+    # create the namespace
+    $ kubectl create ns vineyard-test
+    # get all injected resources
+    $ vineyardctl inject -f pod.yaml | kubectl apply -f -
+    pod/vineyard-sidecar-etcd-0 created
+    service/vineyard-sidecar-etcd-0 created
+    service/vineyard-sidecar-etcd-service created
+    service/vineyard-sidecar-rpc created
+    pod/vineyard-showcase created
+
+
+Then you can get the logs of the consumer containers as follows.
+
+.. code:: bash
+
+    # get the logs of the consumer container
+    $ kubectl logs -f vineyard-showcase -n test -c consumer
+    A   -30.168469
+    B   -19.269489
+    C     6.332533
+    D    -9.714950
+    dtype: float64
+    (1, 1.2345000505447388, 'xxxxabcd')
+
+
+Data sharing between different workloads
+----------------------------------------
```

Comment:
different workloads -> different pods

Reply:
Producer and consumer are different workloads, even when they are placed in the same pod.